volatility/framework/plugins/windows/getsids.py

```diff
@@ -0,0 +1,257 @@
+  'S-1-5-86-615999462-62705297-2911207457-59056572-3668589837': 'WMI (Network Service)',
+  'S-1-5-1000': 'Other Organization',
+  'S-1-16-0': 'Untrusted Mandatory Level',
+  'S-1-16-4096': 'Low Mandatory Level',
+  'S-1-16-8192': 'Medium Mandatory Level',
+  'S-1-16-8448': 'Medium Plus Mandatory Level',
+  'S-1-16-12288': 'High Mandatory Level',
+  'S-1-16-16384': 'System Mandatory Level',
+  'S-1-16-20480': 'Protected Process Mandatory Level',
+  'S-1-16-28672': 'Secure Process Mandatory Level',
+  'S-1-5-21-0-0-0-496': 'Compounded Authentication',
+  'S-1-5-21-0-0-0-497': 'Claims Valid',
+  'S-1-5-32-575': 'RDS Remote Application Services',
+  'S-1-5-32-576': 'RDS Endpoint Servers',
+  'S-1-5-32-577': 'RDS Management Servers',
+  'S-1-5-32-578': 'Hyper-V Admins',
+  'S-1-5-32-579': 'Access Control Assistance Ops',
+  'S-1-5-32-580': 'Remote Management Users',
+  'S-1-5-65-1': 'This Organization Certificate (Kerberos PAC)',
+  'S-1-5-84-0-0-0-0-0': 'Usermode Drivers',
+  'S-1-5-113': 'Local Account',
+  'S-1-5-114': 'Local Account (Member of Administrators)',
+  'S-1-5-1000': 'Other Organization',
+  'S-1-15-2-1': 'Application Package Context',
+  'S-1-18-1': 'Authentication Authority Asserted Identity',
+  'S-1-18-2': 'Service Asserted Identity',
+}
+
+
+class GetSIDs(interfaces.plugins.PluginInterface):
+    """Print the SIDs owning each process"""
+
+    _version = (1, 0, 0)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._protect_values = None
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        # Since we're calling the plugin, make sure we have the plugin's requirements
+        return [requirements.TranslationLayerRequirement(name = 'primary',
+                                                         description = 'Memory layer for the kernel',
+                                                         architectures = ["Intel32", "Intel64"]),
+                requirements.SymbolTableRequirement(name = "nt_symbols", description = "Windows kernel symbols"),
+                requirements.ListRequirement(name = 'pid',
+                                             description = 'Filter on specific process IDs',
+                                             element_type = int,
+                                             optional = True),
+                requirements.PluginRequirement(name = 'pslist', plugin = pslist.PsList, version = (1, 0, 0)),
+                ]
+
+    def lookup_user_sids(self) -> Dict[str, str]:
+        """
+        Enumerate the registry for all the users.
+
+        Returns:
+            An dictionary of {sid: user name}
+        """
+
+        key = "Microsoft\\Windows NT\\CurrentVersion\\ProfileList"
+        val = "ProfileImagePath"
+
+        sids = {}
+        for hive in hivelist.HiveList.list_hives(context = self.context,

```

Comment:
Seems like there should be a `PluginRequirement`/`VersionRequirement` on `hivelist.HiveList`